bin/tx-prover/src/proxy/mod.rs

```diff
@@ -0,0 +1,193 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use once_cell::sync::Lazy;
+use pingora::{
+    http::ResponseHeader,
+    lb::Backend,
+    prelude::*,
+    upstreams::peer::{Peer, ALPN},
+};
+use pingora_core::{upstreams::peer::HttpPeer, Result};
+use pingora_limits::rate::Rate;
+use pingora_proxy::{ProxyHttp, Session};
+use tokio::sync::RwLock;
+use tracing::error;
+
+/// Timeout for the requests
+const TIMEOUT_SECS: Option<Duration> = Some(Duration::from_secs(100));
+/// Maximum number of items per queue
+const MAX_QUEUE_ITEMS: usize = 10;
+
+/// Load balancer that uses a round robin strategy
+pub struct LB(Arc<LoadBalancer<RoundRobin>>);
+
+impl LB {
+    pub fn new(upstreams: LoadBalancer<RoundRobin>) -> Self {
+        Self(Arc::new(upstreams))
+    }
+}
+
+/// Rate limiter
+static RATE_LIMITER: Lazy<Rate> = Lazy::new(|| Rate::new(Duration::from_secs(1)));
+
+/// Maximum amount of request per second per client
+static MAX_REQ_PER_SEC: isize = 5;
```

Comment:
If we ever need to add more parameters like these, maybe it would be a good idea to move them to a config file. Maybe not now as we don't have a lot.

Reply:
Yeah, I think this is a good first approach, but we might also want to create a follow-up issue to discuss config files (and/or maybe adding a small CLI so that it is in line with other Miden projects)

Reply:
Yes, I'm not sure if we need it yet and in case we do if it should be implemented using TOML files like in the other miden-related binaries or using something simpler like env variables. I would appreciate your take on this

Reply:
I think this should be fine for now, and on a follow up we can implement config files and (or maybe even on a third PR) a small CLI. cc @bobbinth just in case